config/configManager.go

```diff
@@ -0,0 +1,195 @@
+}
+
+// Returns created referer store, verifier, policyprovider objects from config
+func createFromConfig(cf Config) ([]referrerstore.ReferrerStore, []verifier.ReferenceVerifier, policyprovider.PolicyProvider, error) {
+	stores, err := sf.CreateStoresFromConfig(cf.StoresConfig, GetDefaultPluginPath())
+
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to load store from config")
+	}
+	logrus.Infof("stores successfully created. number of stores %d", len(stores))
+
+	verifiers, err := vf.CreateVerifiersFromConfig(cf.VerifiersConfig, GetDefaultPluginPath())
+
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to load verifiers from config")
+	}
+
+	logrus.Infof("verifiers successfully created. number of verifiers %d", len(verifiers))
+
+	policyEnforcer, err := pf.CreatePolicyProviderFromConfig(cf.PoliciesConfig)
+
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to load policy provider from config")
+	}
+
+	logrus.Infof("policies successfully created.")
+
+	return stores, verifiers, policyEnforcer, nil
+}
+
+func reloadExecutor(configFilePath string, executor *ef.Executor) error {
+	cf, err := Load(configFilePath)
+
+	if err != nil {
+		logrus.Warnf("failed to load from config file , err: %v", err)
+		return err
+	}
+
+	stores, verifiers, policyEnforcer, err := createFromConfig(cf)
+
+	if err != nil {
+		logrus.Warnf("failed to store/verifier/policy objects from config, no updates loaded. err: %v", err)
+		return err
+	}
+
+	if configHash != cf.FileHash {
+		executor.ReloadAll(stores, verifiers, policyEnforcer, &cf.ExecutorConfig)
+		configHash = cf.FileHash
+		logrus.Infof("configuration file has been updated, reloading executor succeeded")
+	} else {
+		logrus.Infof("no change found in config file, no executor update needed")
+	}
+	return nil
+}
+
+// Setup a watcher on file at configFilePath, reload executor on file change
+func watchForConfigurationChange(configFilePath string, executor *ef.Executor) error {
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		errors.Wrap(err, "new file watcher on configuration file failed ")
+	}
+
+	// setup for loop to listen for events
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					logrus.Warnf("no longer watching configuration file changes, file watcher event channel closed")
+					return
+				}
+
+				logrus.Infof("file watcher event detected %v", event)
+
+				// In a cluster scenario, a configMap will recreate the config file
+				// after the remove event, the watcher will also be removed
+				// since a watcher on a non existent file is not supported, we sleep until the file exist add the watcher back
+				if event.Name == configFilePath && event.Op&fsnotify.Remove == fsnotify.Remove {
+					logrus.Infof("config remove event detected")
+
+					_, err := os.Stat(configFilePath)
+					for err != nil {
+						logrus.Infof("config file does not exist yet, sleeping again")
```

Comment:
Is it polling forever or do we need a timeout?

Reply:
added a 60sec timeout, from my testing, once the remove event has been detected, the files gets recreated immediately